ui/packages/consul-ui/app/routing/route.js

```diff
@@ -107,4 +119,30 @@ export default class BaseRoute extends Route {
       return params;
     }
   }
+
+  @action
+  async replaceWith(routeName, obj) {
+    await Promise.resolve();
+    let params = [];
+    if (typeof obj === 'string') {
+      params = [obj];
+    }
+    if (typeof obj !== 'undefined' && !Array.isArray(obj) && typeof obj !== 'string') {
+      params = Object.values(obj);
+    }
+    return super.replaceWith(routeName, ...params);
+  }
+
+  @action
+  async transitionTo(routeName, obj) {
+    await Promise.resolve();
+    let params = [];
+    if (typeof obj === 'string') {
+      params = [obj];
+    }
+    if (typeof obj !== 'undefined' && !Array.isArray(obj) && typeof obj !== 'string') {
+      params = Object.values(obj);
+    }
+    return super.transitionTo(routeName, ...params);
+  }
```

Comment:
Make transitionTo and replaceWith available to the `route-action` helper, along with a micro-tick to avoid any ember recalculation/reevaluation console errors.